packages/rest/src/body-parsers/body-parser.json.ts

```diff
@@ -36,6 +38,9 @@ export class JsonBodyParser implements BodyParser {
 
   async parse(request: Request): Promise<RequestBody> {
     let body = await invokeBodyParserMiddleware(this.jsonParser, request);
+
+    if (!_.isPlainObject(body)) throw RestHttpErrors.invalidRequestBody();
```

Comment:
Rejecting non-object body unconditionally is not desired. It should be perfect fine to expose an API to accept json array in the body, for example, an array of deals to be added.